packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2748,7 +2747,7 @@ class _FloatingCursorPainter extends RenderEditablePainter {
 
     final Color? floatingCursorColor = this.caretColor?.withOpacity(0.75);
     // Floating Cursor.
-    if (floatingCursorRect == null || floatingCursorColor == null || !shouldPaint)
+    if (floatingCursorRect == null || floatingCursorColor == null)
```

Comment:
it seems like that this is not the original PR changed, am I missing something?

Reply:
I understand now , it's the conflict PR logic.

Reply:
Yeah I had to resolve some merge conflicts in another RenderEditable PR, this is one of the places I had to change to accommodate the caret fix. See the description section